### DISCUSSION TITLE
Key/Velocity scaling as motion sequence / mod source

### DISCUSSION THREAD

**Discussion by juanjo1138** [bangcorrupt/hacktribe#85] [Ideas]
The height of the note-keyboard affects the opening of the filter.

It could be another type of modulation, although **for me it is so essential that I would put it in the first half of the travel of the 'EG Int' control**.

(Or some other way to make it always available without losing the 'EG Int' and without taking up the only available modulation in this…)

(Who uses the negative envelope in the filter? 😛 )

I find 'note height' a very important modulation source, present in practically all synths, and should be able to be applied not only to the filter, but to oscillator parameters, etc.

Too bad that on the electribe you can only select one modulation source at a time…

**Comment by bangcorrupt** [comment]
Key scaling could be possible by copying the note numbers to a motion sequence.  I'd like to combine this with the depth factor of the groove templates, along with separate depth controls for each parameter in the groove templates.  For now, I can probably write a script that copies scaled key or velocity data to a motion sequence.

> Who uses the negative envelope in the filter?

I use this a lot with a high pass filter on snares and other percussion.  Fading the low end out of the tail can make drums sound more lively and realistic.  It can also clean up the mix at busy frequencies.

**Reply by juanjo1138** [reply]
> Key scaling could be possible by copying…

You answered when the title was, I don't know why _"Key/Velocity scaling as motion sequence"_.
There must have been an error (I'm new to this forum) but the title of my post should be _"The height of the note-keyboard affects the opening of the filter."_

I have corrected it and now your answer about "Key scalling" will seem to make little sense. Sorry!

**Reply by bangcorrupt** [reply]
I edited the title to make it more general.  Key and velocity scaling will achieve what you're asking for.

**Reply by juanjo1138** [reply]
> I edited the title to make it more general. Key and velocity scaling will achieve what you're asking for.

Sorry, I hadn't read you and was changing it again!
How embarrassing…

**Reply by juanjo1138** [reply]
> Who uses the negative envelope in the filter?

Yes, it was a bit of a joke, I also use the negative envelope, but I wanted to say that the influence of the keyboard-note seems to me to be a very (more?) important modulation.

**Reply by juanjo1138** [reply]
However, I don't quite understand how your explanation about using it in relation to motion sequence would work…
What I say should work in real time, while tapping on the keypad and/or an external keyboard.

Another thing is that the resulting filter height can be recorded in the 'normal' motion sequence of the filter...

**Reply by bangcorrupt** [reply]
> should work in real time, while tapping on the keypad and/or an external keyboard

This might happen, if I can modify modulation sources.  It could work well if played live as a modulation source but also recorded to a motion sequence, freeing up the modulation slot afterward.  I can enable some more modulation destinations (Resonance, EG Int, Attack, Decay) but they already exist in the firmware.

**Reply by juanjo1138** [reply]
I wonder if you are working alone on this project and how complex it is to understand and modify the behavior of the code.
If you got complete control you could do so many things that Korg should have done... and I,  as many people I think, would certainly be willing to buy it...

**Reply by bangcorrupt** [reply]
As far as I know I'm the only one working on this.  I can't share information about the internals of the firmware as it is copyrighted, making it difficult to collaborate.

I aim to create an open-source firmware everyone can collaborate on. I need to learn a lot more about how the different processors communicate before I can get a basic board support package together.

The patch will always be free (as in beer) and I can't sell pre-patched firmware due to copyright.  If you want to support the project financially you are very welcome to [become a sponsor](https://github.com/sponsors/bangcorrupt), see [How to support hacktribe](https://github.com/bangcorrupt/hacktribe/discussions/63) for more details.
